apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/RetryHandler.cs

```diff
@@ -37,46 +37,64 @@ internal sealed class RetryHandler : IHttpUnsuccessfulResponseHandler
         internal const string InvocationIdHeaderPart = "gccl-invocation-id";
         internal const string AttemptCountHeaderPart = "gccl-attempt-count";
 
-        private static readonly int[] s_retriableErrorCodes =
-        {
-                408, // Request timeout
-                429, // Too many requests
-                500, // Internal server error
-                502, // Bad gateway
-                503, // Service unavailable
-                504 // Gateway timeout
-        };
-        private static readonly RetryHandler s_instance = new RetryHandler();
+        private double _backoffMultiplier;
+        private TimeSpan _initialBackoff;
+        private TimeSpan _maxBackoff;
+        private RetryPredicate _retryPredicate;
 
         private RetryHandler() { }
```

Comment:
Instead of keeping it parameterless, we can pass the parameters that you receive in `MarkAsRetriable` and call this constructor, rather than doing it inline.